src/software/embedded/robot_localizer.cpp

```diff
@@ -0,0 +1,173 @@
+#include "robot_localizer.h"
+#include "shared/constants.h"
+
+#include <iostream>
+
+#include "software/util/scoped_timespec_timer/scoped_timespec_timer.h"
+
+void RobotLocalizer::step(const AngularVelocity& targetAcceleration) {
+    clock_gettime(CLOCK_MONOTONIC, &current_time_);
+    timespec diff = current_time_;
+    ScopedTimespecTimer::timespecDiff(&current_time_, &last_step_, &diff);
+    clock_gettime(CLOCK_MONOTONIC, &last_step_);
+    double deltaTimeSeconds = (double)diff.tv_sec + (double)diff.tv_nsec * SECONDS_PER_NANOSECOND;; // time since last step in seconds
+    FilterStep thisStep;
+    Predict thisPredict;
+    thisStep.birthday = current_time_;
+    thisStep.pre_mean = filter_.x;
+    thisStep.pre_covariance = filter_.P;
+    filter_.F << 1, deltaTimeSeconds, deltaTimeSeconds * deltaTimeSeconds * 0.5,
+            0, 1,                deltaTimeSeconds,
+            0, 0,                1;
+    double timeSquared = deltaTimeSeconds * deltaTimeSeconds;
+    double timeCubed = timeSquared * deltaTimeSeconds;
+    double timeFourth = timeCubed * deltaTimeSeconds;
+    filter_.Q << timeFourth/4, timeCubed/3,      timeSquared/2,
+                 timeCubed/3,  timeSquared,      deltaTimeSeconds,
+                 timeSquared/2,deltaTimeSeconds, 1;
+    filter_.Q *= process_noise_variance_;
+    Eigen::Matrix<double, 1, 1> u;
+    u << targetAcceleration.toRadians() * deltaTimeSeconds; // change in angular velocity due to control input
+    thisPredict.F = filter_.F;
+    thisPredict.Q = filter_.Q;
+    thisPredict.B = filter_.B;
+    thisPredict.u = u;
+    thisStep.prediction = std::optional<Predict>(thisPredict);
+    history.push_front(thisStep);
+    filter_.predict(u);
```

Comment:
code conventions: https://github.com/UBC-Thunderbots/Software/blob/master/docs/code-style-guide.md#names-and-variables

please use snake_case for variables